packages/cli/src/commands/init/utils/repo.ts

```diff
@@ -24,74 +24,89 @@ const isKeyof = <T>(val: T) => {
+    console.log(chalk.green(' - npm run storybook'));
+}
 
-    try {
-        let repoRef: BranchRef | null = null;
-        if (designSystemOptions['ds-framework'] === 'angular') {
-            repoRef = REPOS.angular;
-        } else if (designSystemOptions['ds-framework'] === 'react') {
-            const reactStyle = designSystemOptions['ds-react-style'];
-            if (!isReactStyle(reactStyle)) { throw new Error(`Expected a supported react style, got: ${ reactStyle }`); }
-            repoRef = REPOS.react[reactStyle];
-        }
+export const logFailure = (error: any) => {
+    console.log(chalk.red('Couldn\'t clone the repo.'));
+    console.log(chalk.red(error?.message));
+}
 
-        if (!repoRef) {
-            logger.info('coming soon... 😉');
-            return false;
-        }
+export const selectRepo = (designSystemOptions: any) => {
+    let repoRef: BranchRef | null = null;
 
-        const dir = path.join(process.cwd(), designSystemOptions['ds-name']);
+    if (designSystemOptions['ds-framework'] === 'angular') {
+        repoRef = REPOS.angular;
+    } else if (designSystemOptions['ds-framework'] === 'react') {
+        const reactStyle = designSystemOptions['ds-react-style'];
+        if (!isReactStyle(reactStyle)) { throw new Error(`Expected a supported react style, got: ${ reactStyle }`); }
+        repoRef = REPOS.react[reactStyle];
+    }
+
+    if (!repoRef) {
+        logger.info('coming soon... 😉');
+        return false;
+    }
+    return repoRef;
+}
 
-        const definitiveRef:BranchRef = repoRef;
-        const gitSetup = [
-            {
-                title: 'Download',
-                task: (_ctx:any,task:any) => git.clone({
-                    fs,
-                    dir,
-                    http: gitHttp,
-                    url: `https://github.com/${ definitiveRef.repo }`,
-                    depth: 1,
-                }).catch((err)=>{
-                    fsExtra.removeSync(dir);
-                    task.skip("Git failed.")
-                    throw new Error(err);
-                })
-                
-            },
-            {
-                title: 'Checkout the branch',
-                task: () => git.checkout({
-                    fs,
-                    dir,
-                    ref: definitiveRef.branch,
-                })
-                
-            }
-        ]
-        
+export const configureGitSetup = (dir: any, ref: any, clone: any, checkout: any, removeDir: any): { run: () => Promise<void> } => {
+    const gitSetup = [
+        {
+            title: 'Clone the repo',
+            task: (_ctx: any, task: any) => clone({
+                fs,
+                dir,
+                http: gitHttp,
+                url: `https://github.com/${ref.repo}`,
+                depth: 1,
+            }).catch((err: any) => {
+                removeDir(dir);
+                task.skip("Git failed.")
+                throw new Error(err);
+            })
```

Comment:
Are there types for the arguments in `configureGitSetup`? Might make it easier when we need to call the function and knowing the types beforehand. Other than that it looks good

Reply:
`configureGItSetup` is the way that I exported the usage of isomoprhic-git function (`git.clone`, `git,checkout`) from `cloneRepo` function to increase readability and allow us to test, nothing else will call this function. I didn't add types because they are not actually our variables, isomorphic-git uses them